components/scream/src/physics/share/physics_saturation_impl.hpp

```diff
@@ -11,27 +11,50 @@ namespace physics {
  * this file, #include physics_functions.hpp instead.
  */
 
+template <typename S, typename D>
+KOKKOS_FUNCTION
+void  Functions<S,D>
+::check_temperature(const Spack& t_atm, const char* func_name, const Smask& range_mask)
+{
+
+  //"range_mask" masks out packs which are padded with uninitialized values    
+  
+  /*Future work:
+  Make error message dynamic so that it tells the user the function name (func_name) from where
+  this error is coming from. Currently func_name is not used in this function */
+
+  const auto is_neg_t_atm = (t_atm <= 0) && range_mask; //find out if there are any negative temperatures in the pack
+  if (is_neg_t_atm.any()){
+    scream_kerror_msg("Error! Temperature has <= 0 values.\n"); //exit with an error message
```

Comment:
If you use scream_require_msg, you could combine the mask check and error on the same line.

Reply:
When I use `scream_require_msg`, I get the following error:
```
error: device code does not support exception handling
```
`scream_kerror_msg` takes just one argument.

Reply:
@jgfouca suggested me to use scream_krequire_msg (the device version of this function)